tests/coroutines-mt/src/appleTest/kotlin/macos/app/MainTest.kt

```diff
@@ -26,7 +24,7 @@ class MainTest {
   fun coroutinesMtCanWork() = runTest {
     withContext(Dispatchers.Default) {
       println("Dispatchers.Default: ${currentThreadId()}")
-      withContext(Dispatchers.Main) {
+      withContext(Dispatchers.Main.immediate) {
```

Comment:
I understand `immediate` is required to avoid a deadlock but that feels quite unrepresentative of a real use case.

Especially since we're planning to use `kotlin.runTest {}`, the default dispatcher [should be something like Dispatcher.Main](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-test/kotlinx.coroutines.test/-standard-test-dispatcher.html). In which case something like

```kotlin
  fun coroutinesMtCanWork() = runTest {
    withContext(Dispatchers.Default) {
      // execute GraphQL query here
```

should be enough?